frontend/www/js/omegaup/arena/navigation.test.ts

```diff
@@ -0,0 +1,132 @@
+  input_limit: 10240,
+  karel_problem: true,
+  languages: ['kp', 'kj'],
+  letter: 'A',
+  limits: {
+    input_limit: '10Kibs',
+    memory_limit: '10Kibs',
+    overall_wall_time_limit: '1s',
+    time_limit: '1s',
+  },
+  points: 100,
+  preferred_language: 'kp',
+  problem_id: 1,
+  quality_seal: true,
+  settings: {
+    cases: { easy: { in: '2', out: '4', weight: 1 } },
+    limits: {
+      ExtraWallTime: '1s',
+      OutputLimit: '10Kibs',
+      MemoryLimit: '10Kibs',
+      OverallWallTimeLimit: '1s',
+      TimeLimit: '1s',
+    },
+    validator: {
+      name: 'validator',
+    },
+  },
+  source: 'omegaUp classics',
+  statement: {
+    images: {},
+    sources: {},
+    language: 'kj',
+    markdown: 'some markdown',
+  },
+  title: 'A. Problem',
+  visibility: 2,
+};
+fetchMock.enableMocks();
+fetchMock.mockIf(/^\/api\/.*/, (req: Request) => {
+  if (req.url != '/api/problem/details/') {
+    return Promise.resolve({
+      ok: false,
+      status: 404,
+      body: JSON.stringify({
+        status: 'error',
+        error: `Invalid call to "${req.url}" in test`,
+        errorcode: 403,
+      }),
+    });
+  }
+  return Promise.resolve({
+    status: 200,
+    body: JSON.stringify({
+      ...problemDetails,
+      status: 'ok',
+    }),
+  });
+});
+
+const vueInstance: Vue & {
+  problemInfo: types.ProblemInfo | null;
+  popupDisplayed?: PopupDisplayed;
+  problem: ActiveProblem | null;
+} = new Vue({
+  components: {
+    'omegaup-arena-contest-practice': arena_ContestPractice,
+  },
+  render: function (createElement) {
+    return createElement('omegaup-badge-details', {
+      props: {
+        problemInfo: null,
+        problem: null,
+      },
+    });
+  },
+});
+vueInstance.problemInfo = problemDetails;
+vueInstance.popupDisplayed = PopupDisplayed.RunSubmit;
+const navbarProblems: types.NavbarProblemsetProblem[] = [
+  {
+    acceptsSubmissions: true,
+    alias: 'problem_alias',
+    bestScore: 100,
+    hasRuns: true,
+    maxScore: 100,
+    text: 'A. Problem',
+  },
+  {
+    acceptsSubmissions: true,
+    alias: 'problem_alias_2',
+    bestScore: 80,
+    hasRuns: true,
+    maxScore: 100,
+    text: 'B. Problem 2',
+  },
+];
+
+describe('navigation.ts', () => {
+  describe('navigateToProblem', () => {
+    it('Should change hash when contest alias is declared in practice mode', async () => {
+      const spy = jest.spyOn(navigation, 'setLocationHash');
+      const params: navigation.NavigationRequest = {
+        type: navigation.NavigationType.ForContest,
+        target: vueInstance,
+        runs: [],
+        problems: navbarProblems,
+        problem: navbarProblems[0],
+        contestAlias: 'contest_alias',
+      };
+      navigation.setLocationHash(`#problems/${navbarProblems[0].alias}`);
+      const locationHash = navigation.getLocationHash();
+      expect(spy).toHaveBeenCalled();
+      expect(locationHash).toBe('#problems/problem_alias');
+      navigation.navigateToProblem(params);
```

Comment:
```suggestion
      navigation.navigateToProblem(params);
      expect(spy).toHaveBeenCalledWith('#problems/problem_alias');
```

no hay que llamar `getLocationHash` explícitamente en la prueba. basta con checar que el espía fue llamado con el parámetro esperado.